### ISSUE TITLE
@Optional.Method annotation doesn't work inside anonymous classes

### ISSUE THREAD

**Issue by q3hardcore** [MinecraftForge/FML#602]
Example:
```java
class Test {

    static Test instance = new Test() {
        @Optional.Method(modid="whatever")
        @Override
        void test() {
            throw new RuntimeException();
        }
    };

    void test() {
    }

}
```

**Comment by RainWarrior** [comment]
You need @Optional.InterfaceList on the class itself for it to work. And because you can't add those to anonymous inner classes, you'll have to make it not anonymous.

**Comment by RainWarrior** [comment]
You need @Optional.InterfaceList on the class itself for it to work. And because you can't add those to anonymous inner classes, you'll have to make it not anonymous.

**Comment by LexManos** [comment]
They should work just fine, except for as Fry said, you can't annotate anon-classes themselves.

On top of that, this is what would be considered abuse of the Optional annotation. Its to prevent the functions from existing/crashing the game if there signature references non-existent types.

**Comment by cpw** [comment]
Agree with what both Lex and Rain said. Optional should not be used on anonymous classes. If it's an anonymous class, it's highly likely it won't get classloaded until you're in the method - so traditional "test for mod existence" tricks should work fine.

**Comment by diesieben07** [comment]
In addition this is fixed now: #617.
